File: hw/astudillo/U3/HW18Factory/AbstractFactoryPython/ec/edu/espe/composite/abstractfactory/controller/LinuxFactory.py
"""
Linux Factory implementation

@author PC
"""
from .GUIFactory import GUIFactory
from ec.edu.espe.composite.abstractfactory.model.Button import Button
from ec.edu.espe.composite.abstractfactory.model.Menu import Menu
from ec.edu.espe.composite.abstractfactory.view.LinuxButton import LinuxButton
from ec.edu.espe.composite.abstractfactory.model.LinuxMenu import LinuxMenu


class LinuxFactory(GUIFactory):
    
    def create_button(self):
        return LinuxButton()
    
    def create_menu(self):
        return LinuxMenu()

File: hw/suarez/U3/HW18AbstractFactory/src/ec/edu/espe/model/LinuxFactory.java
package ec.edu.espe.model;

/**
 *
 * @author bernardo
 */

public class LinuxFactory extends GUIFactory {

    @Override
    public Button createButton() {
        return new LinuxButton();
    }

    @Override
    public Menu createMenu() {
        return new LinuxMenu();
    }

}
